File: E_Commerce/E_Commerce/E_Commerce.Dal/Entities/Card.cs
namespace E_Commerce.Dal.Entities;

public class Card
{
    public long CardId { get; set; }
    public long CustomerId { get; set; }
    public string Number { get; set; }
    public string HolderName { get; set; }
    public int ExpirationMonth { get; set; }
    public int ExpirationYear { get; set; }
    public int? Cvv { get; set; }
    public bool SelectedForPayment { get; set; }
    public Customer Customer { get; set; }
}

File: Client/src/app/services/models/Card.ts
import {CustomerModel } from "./Customer";

export class CardModel {
  cardId: number = 0; // long in C# maps to number in TypeScript
  customerId: number = 0;
  number: string = ''; // 'Number' is a reserved keyword in JavaScript/TypeScript, so use 'number' (lowercase)
  holderName: string = '';
  expirationMonth: number = 0;
  expirationYear: number = 0;
  cvv?: number | null; // int? in C# maps to number | null | undefined in TypeScript
  selectedForPayment: boolean = false;
  customer?: CustomerModel; 
}
